.github/workflows/sql-validation.yaml

```diff
@@ -0,0 +1,156 @@
+            else
+              echo "Oracle linting encountered issues, check the report for details"
+            fi
+            echo "Oracle lint report:"
+            cat reports/oracle_lint_report.yaml
+          else
+            echo "No Oracle directory found"
+          fi
+
+      - name: Lint MySQL files
+        run: |
+          echo "Linting MySQL files:"
+          if [ -d "mysql" ]; then
+            echo "MySQL directory found. Contents:"
+            ls -R mysql
+            echo "Running SQLFluff on MySQL files:"
+            sqlfluff lint --dialect mysql --format yaml mysql || true
+            echo "Saving lint results:"
+            if sqlfluff lint --dialect mysql --format yaml mysql > reports/mysql_lint_report.yaml; then
+              echo "MySQL linting completed successfully"
+            else
+              echo "MySQL linting encountered issues, check the report for details"
+            fi
+            echo "MySQL lint report:"
+            cat reports/mysql_lint_report.yaml
+          else
+            echo "No MySQL directory found"
+          fi
+
+      - name: Check SQL syntax
+        run: |
+          echo "Checking SQL syntax:"
+          touch reports/syntax_report.txt
+          find_output=$(find . -name "*.sql")
+          if [ -z "$find_output" ]; then
+            echo "No SQL files found" | tee -a reports/syntax_report.txt
+          else
+            while IFS= read -r file; do
+              echo "Checking $file" | tee -a reports/syntax_report.txt
+              if ! cat "$file" | sqlparse --validate >> reports/syntax_report.txt 2>&1; then
+                echo "Error validating $file" | tee -a reports/syntax_report.txt
+              fi
+              echo "" >> reports/syntax_report.txt
+            done <<< "$find_output"
+          fi
+          echo "Syntax check complete. Report contents:"
+          cat reports/syntax_report.txt
+
+      - name: Generate Markdown Report
+        run: |
+          report_file="reports/report-$(date +'%Y%m%d-%H%M%S').md"
+          echo "# SQL Validation Report" > $report_file
+          echo "## MySQL Linting Results" >> $report_file
+          echo '```yaml' >> $report_file
+          cat reports/mysql_lint_report.yaml >> $report_file
+          echo '```' >> $report_file
+          echo "## Oracle Linting Results" >> $report_file
+          echo '```yaml' >> $report_file
+          cat reports/oracle_lint_report.yaml >> $report_file
+          echo '```' >> $report_file
+          echo "## Syntax Check Results" >> $report_file
+          echo '```' >> $report_file
+          cat reports/syntax_report.txt >> $report_file
+          echo '```' >> $report_file
+
+      - name: Upload report
+        uses: actions/upload-artifact@v3
+        with:
+          name: sql-validation-report
+          path: reports/report-*.md
+          retention-days: 7
+
+  cleanup-old-reports:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: write
+
+    steps:
```

Comment:
if the reports are the artefacts and they have a retention period what is this cleaning up?

Reply:
Ah, yes. GitHub's artifact retention policy applies to individual artifacts, not the overall repository storage. With this, we have a rolling 7-day window of reports whilst preventing long-term accumulation of outdated artifacts. Except we want to keep the [90-days default](https://docs.github.com/en/enterprise-server@3.11/admin/enforcing-policies/enforcing-policies-for-your-enterprise/enforcing-policies-for-github-actions-in-your-enterprise) one.